File: src/controllers/letter_controller.py
# src/controllers/letter_controller.py

# ... existing imports ...
from flask import Blueprint, request, jsonify
from src.services.letter_service import generate_letter_body

letter_bp = Blueprint("letter", __name__)

# ... existing code ...
@letter_bp.route("/letters", methods=["POST"])
def create_letter():
    try:
        input_data = request.get_json(force=True) or {}
        body = generate_letter_body(input_data)
        return jsonify({"body": body}), 200
    except Exception as e:
        # ... existing error handling ...
        return jsonify({"error": str(e)}), 500
# ... existing code ...

File: src/controllers/letterController.ts
// src/controllers/letterController.ts

// ... existing imports ...
import { generateLetterBody, type LetterInput } from "../services/letterService";

// ... existing code ...
export async function createLetter(req: any, res: any) {
  try {
    const input: LetterInput = {
      recipientName: req.body.recipientName,
      subject: req.body.subject,
      keyPoints: req.body.keyPoints, // array of strings
      tone: req.body.tone,           // "formal" | "informal" | "neutral"
      language: req.body.language,   // e.g., "English"
      senderName: req.body.senderName,
    };

    const body = await generateLetterBody(input);
    res.status(200).json({ body });
  } catch (err: any) {
    // ... existing error handling ...
    res.status(500).json({ error: err?.message || "Failed to generate letter body" });
  }
}
// ... existing code ...